### ISSUE TITLE
Possible to eliminate need for SD card ?

### ISSUE THREAD

**Issue by qwertangel** [EliasKotlyar/Xiaomi-Dafang-Hacks#1294]
* exact camera hardware: Wyze Cam V2

### Description
It would be nice to not need a SD card after dafang hacks are installed (have hacks directly flashed to camera). There is another project https://github.com/HclX/WyzeHacks which claims to not need sd card after flashing.

**Comment by jmtatsch** [comment]
Possible yes, worth the effort and the crying of people with bricked cams - no!

**Comment by fonix232** [comment]
It should be relatively straightforward, I agree.

All the initial hack does (before the CFW is copied to the disk) is to modify the boot order. If the CFW was packaged as a squashfs image, it could replace the system partition on the device itself, instead of relying on the SD card.

@jmtatsch I think it would be worthwhile to at least have a guide (and maybe some tested images?) in place for advanced users. Once your hack is in place, since it would boot from the SD card first, a rescue system could be integrated that restores the original FW if anything goes wrong. In fact if you modify the boot order (so that it first tries the internal NAND, then the microSD), it should be relatively straightforward to have a recovery system on the SD card with the latest CFW and OFW images, that launches an SSH daemon and allows the user to log in, and select the appropriate update file (or use a pre-populated config file). This would be more resilient to possible damages and user interventions into the process, allow to boot the CFW right from the camera, and on the long run it would simplify the process of installation (once it's been tested).

I've just gotten a handful of Neos smartcams (Wyze V2, but different software), and I wouldn't mind helping out writing such a system, and testing it. The downside is that any change to the rootfs would require a new build to be launched, and a new image to be hosted - however if you're willing to get into a free CI system (e.g. Travis or Bitrise), I could automate the whole process for you.

Shall I open a separate issue ticket for that? I could go into more detail what the goal is, how it's going to happen, etc.

**Comment by Ozzyminted** [comment]
I would also be interested in getting involved with this. I have looked at
and tried with openfang which was promising. But there is no documentation
and no one seems to be doing anything more with it. And it doesn't seem to
like Neos cameras.
i have tried rebuilding, dissecting it, but its beyond my current level of
skill.

Its been a great experience for me though. Lots of tinkering about and
getting to learn more about it all.

**Comment by fonix232** [comment]
@Ozzyminted the openfang build system actually looks pretty solid - I'm sure it could be easily adopted to build this CFW instead.

**Comment by jmtatsch** [comment]
Openfang is absolutely what you are looking for. I am pretty sure the author will give you hints / help as long as you do the majority of the work. Porting most of the candy from dafang hacks is straightforward.
